src/Component.php

```diff
@@ -65,7 +65,8 @@ public function run(): void
                     $subFinder->in($sourceFile->getPathname())->depth(0);
                     if (!count($subFinder)) {
                         throw new Exception(
-                            "Sliced file '{$sourceFile->getPathname()}' does not contain slices."
```

Comment:
Neměla by to být UserException, když už vylepšuješ tu message? Tohle je "App". Plus teda koukám, že tohle ještě nevyužívá UserException z php-component a nechytá ji to v run.php - to by teda chtělo taky doplnit.

Reply:
jinak na sprintf vs inline ve stringu se asi neshodneme :) Ale to není problém pro review

Reply:
resp. spis by  se ta Exception mela zachytavat tady https://github.com/keboola/processor-create-manifest/blob/master/src/run.php#L15

Reply:
No, je otázka kde všude se vyhazuje. Podle toho.

Reply:
no prave podle toho by mela https://github.com/keboola/processor-create-manifest/blob/master/src/Component.php#L98 
jediny nejisty uziti je tohle https://github.com/keboola/processor-create-manifest/blob/master/src/Component.php#L124

Reply:
No, ale čistě z hlediska konzistence by to mělo být spíš tak, že když už máš UserException, tak by se měla chytat ta. A potenciálně od ní klidně může takhle dědit. Ale optimálně by ten "run.php" měl být šablonovitý copy paste z generátoru a cokoli co si házíš uvnitř a mělo by být user bys měl ošetřit a zabalit do UserException.

Reply:
ano, z hlediska konzistence - vymenit run.php za strandardni, ale pokud to ma byt minimalni zmena, tak prehodit na userexception nebo zachytavat exception

Reply:
OK, co takhle:

* přidat catch na tuhle exception s exit(1)
* použití na L124 změnit na \Exception
* tohle se může mergnout
* připravit další PR, kde se to sladí chytání na verzi z component-generátoru (c&p run.php) s tím, že ty, které jsou teď v catch navíc se chytnou tam kde se vyhazují a obalí UserException

Reply:
jo, az na posledni bod, ve kterym by se stavajici exception mela proste vykopat ven a pouzit php-component user exception

Reply:
Done. https://github.com/keboola/processor-create-manifest/issues/22

Reply:
@tomasfejfar máš nějakej důvod, proč preferovat sprintf oproti inline? rychlej google mi hodil https://r.je/sprintf.html

Reply:
jak jsou tam uvnitr ty uvozovky tak mi prijde taky lepsi ten sprintf, tj.
`'Hi ' . $name . ', you last logged in on ' . $date`
vs.
`'Hi "' . $name . '", you last logged in on '" . $date . '"'`

Reply:
Oba dva přístupy jsou pro SQL stejně špatně :-)

> Ideally you should be using PDO anyway which has named parameters that work similarly to sprintf(). This still suffers from reduced readability (It at least uses named parameters though! No %s meaning the supplied arguments don't need to be in order!) but the security advantages alone far outweigh this. If you're not using PDO you should be using a function which maps all query parameters through the relevant *_escape_string function and work similarly to PDO.

Reply:
nechci hajit sprintf, ale PDO parameters jsou pri ` SELECT %s  FROM ` na dve veci :)

Reply:
no jasně, pokud ten dotaz generuješ, tak je to naprd

Reply:
jj, to je [typické použití](https://github.com/keboola/connection/blob/b4b1f9acf2a05b85eced5b4f34d2005b3d699046/application/modules/core/models/Buckets.php#L89
) a to nemluvím o tom, že jak se to hezky čte tak to demotivuje od escapování ;) 

Třeba Zend 2 Zend\Db má už i UPDATE/DELETE queries skrz PHP API, ale v Zend_Db se bez toho neobjedeme.